addons/sale_purchase/models/sale_order.py

```diff
@@ -288,7 +288,10 @@ def _purchase_service_create(self, quantity=False):
                 ], limit=1)
             if not purchase_order:
                 values = line._purchase_service_prepare_order_values(supplierinfo)
-                purchase_order = PurchaseOrder.create(values)
+                if self.env.user.has_group('base.group_user'):
+                    purchase_order = PurchaseOrder.create(values)
+                else:
+                    purchase_order = PurchaseOrder.with_user(SUPERUSER_ID).create(values)
```

Comment:
@daonamutc1 What is happen? I think should using `sudo()`, it does not change the current user 
```suggestion
                    purchase_order = PurchaseOrder.sudo().create(values)
```

Reply:
We need to change the current user.
Because the current user is user portal or public.
When creating a PO, the system will recognize the creator as the current user and automatically add follower for that user. Then the portal user can see the PO, see the supplier

Reply:
The main problem is the creator is added to the followers automatically, so just don't add him anymore to solve it. You can use `mail_create_nosubscribe` context to do the job instead of this complex solution.

Reply:
Thanks